crates/worker/src/commands/utils.rs

```diff
@@ -0,0 +1,183 @@
+use aws_sdk_dynamodb::error::SdkError;
+use aws_sdk_dynamodb::operation::update_item::UpdateItemError;
+use aws_sdk_dynamodb::types::AttributeValue;
+use aws_sdk_s3::presigning::PresigningConfig;
+use std::path::Path;
+use std::time::Duration;
+use tracing::{error, warn};
+use types::item::{Item, Status, TaskResult};
+use types::{CompilationRequest, ARTIFACTS_FOLDER};
+use uuid::Uuid;
+
+use crate::clients::dynamodb_client::DynamoDBClient;
+use crate::clients::errors::DBError;
+use crate::clients::s3_client::S3Client;
+use crate::commands::compile::{ArtifactData, CompilationInput, CompilationOutput};
+use crate::commands::errors::{CommandResultHandleError, PreparationError};
+use crate::utils::cleaner::AutoCleanUp;
+use crate::utils::lib::{SOL_ROOT, ZKSOLC_VERSIONS};
+
+async fn try_set_compiling_status(
+    db_client: &DynamoDBClient,
+    key: Uuid,
+) -> Result<(), PreparationError> {
+    let db_update_result = db_client
+        .client
+        .update_item()
+        .table_name(db_client.table_name.clone())
+        .key(Item::primary_key_name(), AttributeValue::S(key.to_string()))
+        .update_expression("SET #status = :newStatus")
+        .condition_expression("#status = :currentStatus")
+        .expression_attribute_names("#status", Status::attribute_name())
+        .expression_attribute_values(
+            ":newStatus",
+            AttributeValue::N(u32::from(Status::InProgress).to_string()),
+        )
+        .expression_attribute_values(
+            ":currentStatus",
+            AttributeValue::N(u32::from(Status::Pending).to_string()),
+        )
+        .send()
+        .await;
+
+    match db_update_result {
+        Ok(_) => Ok(()),
+        Err(SdkError::ServiceError(err)) => match err.err() {
+            UpdateItemError::ConditionalCheckFailedException(_) => {
+                error!("Conditional check not met");
+                Err(PreparationError::UnexpectedStatusError(
+                    "Concurrent status change from another instance".into(),
+                ))
+            }
+            _ => Err(DBError::from(SdkError::ServiceError(err)).into()),
+        },
+        Err(err) => Err(DBError::from(err).into()),
+    }
+}
+
+pub(crate) async fn prepare_compile_input(
+    request: &CompilationRequest,
+    db_client: &DynamoDBClient,
+    s3_client: &S3Client,
+) -> Result<CompilationInput, PreparationError> {
+    let zksolc_version = request.config.version.as_str();
+    if !ZKSOLC_VERSIONS.contains(&zksolc_version) {
+        return Err(PreparationError::VersionNotSupported(
+            zksolc_version.to_string(),
+        ));
+    }
+
+    let item = db_client.get_item(request.id.to_string().as_str()).await?;
+    let item: Item = match item {
+        Some(item) => item,
+        None => {
+            error!("No item id: {}", request.id);
+            return Err(PreparationError::NoDBItemError(request.id.to_string()));
+        }
+    };
+
+    match item.status {
+        Status::Pending => {}
```

Comment:
will something be here? if not why not to use `if !matches!(..)`?

Reply:
Done